kolibri/core/content/serializers.py

```diff
@@ -248,3 +252,60 @@ def get_updated_resource(self, instance):
 
     def get_is_leaf(self, instance):
         return instance.kind != content_kinds.TOPIC
+
+
+class ContentDownloadRequestMetadataSerializer(serializers.Serializer):
+
+    title = serializers.CharField()
+    file_size = serializers.IntegerField()
+    learning_activities = serializers.ListField(
+        child=serializers.ChoiceField(learning_activities.choices)
+    )
+
+
+class ContentDownloadRequestSeralizer(serializers.ModelSerializer):
+
+    metadata = ContentDownloadRequestMetadataSerializer()
+
+    class Meta:
+
+        model = ContentDownloadRequest
+        fields = (
+            "id",
+            "contentnode_id",
+            "metadata",
+        )
+
+    def create(self, validated_data):
+        # if there is an existing deletion request, delete the deletion request
+        if "request" in self.context and self.context["request"].user is not None:
+            user = self.context["request"].user
+        else:
+            raise serializers.ValidationError("User must be defined")
+
+        deletion_request = ContentRemovalRequest.objects.filter(
+            contentnode_id=validated_data["contentnode_id"],
+            source_id=user.id,
+        )
+
+        if deletion_request.exists():
+            ContentRemovalRequest.objects.filter(
+                contentnode_id=validated_data["contentnode_id"],
+                source_id=user.id,
+            ).delete()
```

Comment:
You could probably remove the `if deletion_request.exists():` as I don't think a `.delete()` on a queryset will fail if there's nothing to delete, but I think it's fine having it for clarity. Additionally, can reuse `deletion_request` to call the `.delete()`:
```python
if deletion_request.exists():
    deletion_request.delete()
```

Reply:
Yeah, just calling the `delete` method should be sufficient here - as it will be a no op if exists is not true, and saves a query.